driver/src/main/scala/za/co/absa/hyperdrive/driver/SparkIngestor.scala

```diff
@@ -96,10 +105,26 @@ object SparkIngestor extends SparkIngestorAttributes {
   def apply(conf: Configuration): SparkIngestor = {
     ComponentFactoryUtil.validateConfiguration(conf, getProperties)
     val spark = getSparkSession(conf)
+    val terminationMethod = getTerminationMethod(conf)
     val awaitTerminationTimeout = getAwaitTerminationTimeoutMs(conf)
 
-    logger.info(s"Creating ingestor: await termination timeout = '$awaitTerminationTimeout'")
-    new SparkIngestor(spark, awaitTerminationTimeout, conf)
+    logger.info(s"Creating ingestor: termination method = '$terminationMethod', " +
+      s"await termination timeout = '$awaitTerminationTimeout'")
+    new SparkIngestor(spark, terminationMethod, awaitTerminationTimeout, conf)
+  }
+
+  private def getTerminationMethod(conf: Configuration): TerminationMethod = {
+    ConfigUtils.getOrNone(KEY_TERMINATION_METHOD, conf) match {
+      case Some(name) => parseTerminationMethod(name)
+      case None => AwaitTermination
```

Comment:
In initial impl there was ProcessAllAvailable

Reply:
+ in read me is default for ingestor.spark.termination.method ProcessAllAvailable

Reply:
Ah yes. The doc is wrong. Default should be awaitTermination to not change the behaviour from the current version